cmd/microcloud/status.go

```diff
@@ -0,0 +1,444 @@
+}
+
+// Symbol returns a word representing the StatusLevel, color coded.
+func (s StatusLevel) String() string {
+	switch s {
+	case Success:
+		return style.SuccessColor("HEALTHY")
+	case Warn:
+		return style.WarningColor("WARNING")
+	case Error:
+		return style.ErrorColor("ERROR")
+	}
+
+	return ""
+}
+
+type cmdStatus struct {
+	common *CmdControl
+}
+
+func (c *cmdStatus) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Deployment status with configuration warnings",
+		RunE:  c.Run,
+	}
+
+	return cmd
+}
+
+func (c *cmdStatus) Run(cmd *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return cmd.Help()
+	}
+
+	cloudApp, err := microcluster.App(microcluster.Args{StateDir: c.common.FlagMicroCloudDir})
+	if err != nil {
+		return err
+	}
+
+	status, err := cloudApp.Status(context.Background())
+	if err != nil {
+		return fmt.Errorf("Failed to get MicroCloud status: %w", err)
+	}
+
+	if !status.Ready {
+		return fmt.Errorf("MicroCloud is uninitialized, run 'microcloud init' first")
+	}
+
+	cfg := initConfig{
+		autoSetup: true,
+		bootstrap: false,
+		common:    c.common,
+		asker:     &c.common.asker,
+		systems:   map[string]InitSystem{},
+		state:     map[string]service.SystemInformation{},
+	}
+
+	cfg.name = status.Name
+	cfg.address = status.Address.Addr().String()
+
+	services := []types.ServiceType{types.MicroCloud, types.LXD}
+	optionalServices := map[types.ServiceType]string{
+		types.MicroCeph: api.MicroCephDir,
+		types.MicroOVN:  api.MicroOVNDir,
+	}
+
+	services, err = cfg.askMissingServices(services, optionalServices)
+	if err != nil {
+		return err
+	}
+
+	// Instantiate a handler for the services.
+	sh, err := service.NewHandler(status.Name, status.Address.Addr().String(), c.common.FlagMicroCloudDir, c.common.FlagLogDebug, c.common.FlagLogVerbose, services...)
+	if err != nil {
+		return err
+	}
+
+	cloudClient, err := sh.Services[types.MicroCloud].(*service.CloudService).Client()
+	if err != nil {
+		return err
+	}
+
+	// Query the status API for the cluster.
+	statuses, err := client.GetStatus(context.Background(), cloudClient)
+	if err != nil {
+		return err
+	}
+
+	// compile all warning messages.
+	warnings := compileWarnings(cfg.name, statuses)
+
+	// Print the warning summary, and all warnings.
+	fmt.Println("")
+	fmt.Printf(" %s: %s\n", style.TextColor("Status"), warnings.Status().String())
+	fmt.Println("")
+	for _, w := range warnings {
+		fmt.Printf(" %s %s %s\n", style.TableBorderColor("┃"), w.Level.Symbol(), w.Message)
+	}
+
+	fmt.Println("")
+
+	headers := []string{"Name", "Address", "OSD Disks", "MicroCeph Services", "MicroOVN Services", "Status"}
+
+	var localStatus types.Status
+	for _, s := range statuses {
+		if s.Name == cfg.name {
+			localStatus = s
+		}
+	}
+
+	// Format and colorize cells of the table.
+	rows := make([][]string, 0, len(statuses))
+	for _, s := range statuses {
+		rows = append(rows, formatStatusRow(localStatus, s))
+	}
+
+	t := table.New()
```

Comment:
Worth having all of these calls in a helper? I wonder when we create more tables that we can save some repetitive tasks.